### PR TITLE
Typos

### PR DESCRIPTION
**Pull request by petervdonovan**
While I was reading the book I found some typos. I tried to limit my changes to those that actually improve clarity and kept meaning to do a second read-through to make sure that all my suggestions actually were necessary. However, to keep my branch from falling too far behind, it seems appropriate to open this PR now.

### CODE REVIEW

File: chisel-book.tex [hunk 1]
```diff
@@ -4360,19 +4361,20 @@ \section{Lightweight Components with Functions}
 
 \shortlist{code/components_fn_2delay.txt}
 
-\noindent Again, this is a too short example to be useful, as \code{RegNext()}
+\noindent Again, this is a too short an example to be useful, as \code{RegNext()}
```
Comment (petervdonovan) [inline chisel-book.tex:4364]:
```suggestion
\noindent Again, this is too short an example to be useful, as \code{RegNext()}
```

File: chisel-book.tex [hunk 2]
```diff
@@ -5581,9 +5583,8 @@ \subsection{Pipelined Handshake}
 Also have a write timing diagram. And find a better font for the diagram to have nicer
 rendering in the SVG.}
 
-Here we define a simple pipelined handshake bus protocol that avoids the single
-cycle combinational loop and fits better for modern SoC designs.
- A read or write command are signaled by an assertion of \code{rd} or \code{wr}
+Here we define a simple pipelined handshake bus protocol that avoids the single-cycle combinational loop and fits better for modern SoC designs.
```
Comment (petervdonovan) [inline chisel-book.tex:5586]:
```suggestion
Here we define a simple pipelined handshake bus protocol that avoids the single-cycle combinational loop and fits better with modern SoC designs.
```

File: chisel-book.tex [hunk 3]
```diff
@@ -5594,17 +5595,17 @@ \subsection{Pipelined Handshake}
 
 Figure~\ref{fig:bus_pipe_ack} shows such a bus protocol that does not need
 a combinational reaction of the peripheral device. The request from the processor
-is only a single clock cycle long. The address bus and the read signal does not need to be driven
-till the acknowledgment. Compared to the former protocol, the \code{ack} signal needs to
+is only a single clock cycle long. The address bus and the read signal do not need to be driven
+until the acknowledgment. Compared to the former protocol, the \code{ack} signal needs to
 be valid (low or high)
 earliest one clock cycle after the \code{rd} command, in clock cycle 3.
```
Comment (petervdonovan) [inline chisel-book.tex:5601]:
```suggestion
no earlier than one clock cycle after the \code{rd} command, in clock cycle 3.
```

File: chisel-book.tex [hunk 4]
```diff
@@ -5642,9 +5643,9 @@ \subsection{Example IO Device}
 \subsection{Memory Mapped Devices}
 \index{Memory mapped IO}
 
-In our example system all devices, being it memory or IO devices, are connected to a shared
+In our example system all devices, being it memory or IO devices, are connected to shared
```
Comment (petervdonovan) [inline chisel-book.tex:5646]:
```suggestion
In our example system all devices, whether they be memory or IO devices, are connected to shared
```

File: chisel-book.tex [hunk 5]
```diff
@@ -5769,29 +5770,29 @@ \subsection{Wishbone}
...
 that case the address decoding in the slave can be performed in the
-same cycle as the address is registered. A similar issue, with
+same cycle as when the address is registered. A similar issue, with
 respect to the master, exists for the output data from the slave: As
 it is only valid for a single cycle the data has to be registered by
 the master when the master is not reading it immediately. Therefore,
 the slave should keep the last valid data at its output even when
 the Wishbone strobe signal (\emph{wb.stb}) is not assigned anymore.
-Holding the data in the slave is usually \emph{for free} from the
+Holding the data in the slave is usually \emph{free} with respect to
```
Comment (petervdonovan) [inline chisel-book.tex:5791]:
```suggestion
Holding the data in the slave is usually \emph{free} in terms of
```

File: chisel-book.tex [hunk 6]
```diff
@@ -5915,28 +5916,28 @@ \section{Debugging}
 \section{Testing in Chisel}
 
 ChiselTest is based on ScalaTest. Therefore, we can run all tests with a simple \code{sbt test}.
-ScalaTest also supports multithreaded testing out of the box meaning that if you have more 
-test classes in your project, they can run in parallel. Additionally, you can make use of 
+ScalaTest also supports multithreaded testing out of the box meaning that if you have more
```
Comment (petervdonovan) [inline chisel-book.tex:5919]:
```suggestion
ScalaTest also supports multithreaded testing out of the box, so if you have multiple
```

File: chisel-book.tex [hunk 7]
```diff
@@ -5915,28 +5916,28 @@ \section{Debugging}
...
 we call \code{step()} on the DUT's implicit clock port.
 
 You can define a test within the \code{test()} function which takes the module to test as
 parameter. As an example, the following tester tests a few inputs to  a BCD table:
 
 \shortlist{code/basic_chiseltest.txt}
 
-Alternatively, you can use the \code{behavior of "module name"} syntax to 
-refer to the module with \code{it}. This is useful when you have several tests for a 
+Alternatively, you can use the \code{behavior of "module name"} syntax to
```
Comment (petervdonovan) [inline chisel-book.tex:5939]:
```suggestion
Alternatively, you can use the \code{behavior of ``module name''} syntax to
```

File: chisel-book.tex [hunk 8]
```diff
@@ -5962,33 +5963,33 @@ \section{Testing in Chisel}
...
 the test will hang in an endless loop. A robuster read function shall contain a timeout for the \code{ack}
 polling. Finally, we read the data from \code{rdData} with \code{peekInt()} to read a Scala integer
 value (concrete a \code{BigInt} to express integers of any size).
 
 The \code{write} function takes an address and the data parameters as Scala \code{Int}.
-Similar to the read function, the values are poked into the device, the clock advanced one
-clock cycle and then the write signal deasserted. Also here we wait in an endless loop that
-\code{ack} becomes true.
+Similar to the read function, the values are poked into the device, the clock is advanced by one
+clock cycle, and then the write signal deasserted. Also here we wait in an endless loop for
```
Comment (petervdonovan) [inline chisel-book.tex:5974]:
```suggestion
clock cycle, and then the write signal deasserted. Here we also wait in an endless loop for
```

File: chisel-book.tex [hunk 9]
```diff
@@ -5962,33 +5963,33 @@ \section{Testing in Chisel}
...
+\code{ack} to become true.
 
-With those three functions available, we can write better readable tests in less lines of code.
-This testing code covers already more cases than the original bit-banging
-tester.\footnote{It happened that I had an off by one error (\code{until 3} instead of \code{until 4})
-in the counter device, that I found only with the second, more comprehensive test.}
+With those three functions available, we can write more readable tests in fewer lines of code.
```
Comment (petervdonovan) [inline chisel-book.tex:5977]:
```suggestion
With those three functions available, we can write more readable tests with fewer lines of code.
```

File: chisel-book.tex [hunk 10]
```diff
@@ -5962,33 +5963,33 @@ \section{Testing in Chisel}
...
-run. The easiest way to achieve this and still have to run only a single SBT command 
+If you have a large test suite, you may wish to run only a subset of your tests
+as part of a \myref{https://en.wikipedia.org/wiki/Continuous_integration}{continuous integration}
+run. The easiest way to achieve this and still have to run only a single SBT command
 is by tagging your tests.
 
 \shortlist{code/tag_test.txt}
 
-\noindent Per default, all tests are run by using \code{sbt test} or \code{sbt testOnly *}. 
+\noindent Per default, all tests are run by using \code{sbt test} or \code{sbt testOnly *}.
```
Comment (petervdonovan) [inline chisel-book.tex:5992]:
```suggestion
\noindent By default, all tests are run using \code{sbt test} or \code{sbt testOnly *}.
```

File: chisel-book.tex [hunk 11]
```diff
@@ -6004,69 +6005,69 @@ \section{Testing in Chisel}
...
 subsequent threads.
 
 \section{Simulator Backends}
 \index{Verilator}
 \index{VCS}
 
-Per default, tests written with ChiselTest are run by the Treadle simulation backend.
-The benefit of Treadle is the quick startup time and no additional tools need to be installed.
+By default, tests written with ChiselTest are run by the Treadle simulation backend.
+The benefit of Treadle is the quick startup time and the fact that no additional tools need to be installed.
```
Comment (petervdonovan) [inline chisel-book.tex:6038]:
```suggestion
The benefits of Treadle are its quick startup time and the fact that it does not require any additional tools to be installed.
```

File: chisel-book.tex [hunk 12]
```diff
@@ -6004,69 +6005,69 @@ \section{Testing in Chisel}
...
-Verilator is open-source, we will use it for the examples presented in this section. Note 
-that in all cases, VCS can be be used as a an alternative to Verilator.
+\myref{https://www.veripool.org/wiki/verilator}{Verilator} and
+\myref{https://www.synopsys.com/verification/simulation/vcs.html}{Synopsys VCS}. Because
+Verilator is open-source, we will use it for the examples presented in this section. Note
+that in all cases, VCS can be be used as an alternative to Verilator.
 
-Switching to a different backend is simply a matter of adding another annotation to the 
+Switching to a different backend is simply a matter of adding another annotation to the
 \code{withAnnotations} call as shown in the Waveforms section. To use Verilator, add following
```
Comment (petervdonovan) [inline chisel-book.tex:6049]:
```suggestion
\code{withAnnotations} call as shown in the Waveforms section. To use Verilator, add the following
```

File: chisel-book.tex [hunk 13]
```diff
@@ -6634,26 +6635,26 @@ \subsection{Regular Publishing}
 
 \section{Contributing to Chisel}
 
-The following is an advanced topic and I propose you first start following the discussions
-of the Chisel project issues on GitHub, before creating your first pull request (PR).
+The following is an advanced topic, and I propose you first start following the discussions
+of the Chisel project issues on GitHub before creating your first pull request (PR).
```
Comment (petervdonovan) [inline chisel-book.tex:6639]:
```suggestion
The following is an advanced topic, and I propose that you first start following the discussions
of the Chisel project issues on GitHub before you create your first pull request (PR).
```

File: chisel-book.tex [hunk 14]
```diff
@@ -6634,26 +6635,26 @@ \subsection{Regular Publishing}
...
-firrt/Chisel change you might need to also fork and clone firrtl}
+FIRRTL/Chisel change, you might need to also fork and clone \code{firrtl}.}
 In our example, we change \code{chisel3}, and the clone command for my local fork is:
 
 \begin{verbatim}
 $ git clone git@github.com:schoeberl/chisel3.git
 \end{verbatim}
 
-To compile Chisel 3 and publish as a local library execute:
+To compile Chisel 3 and publish it as a local library execute:
```
Comment (petervdonovan) [inline chisel-book.tex:6657]:
```suggestion
To compile Chisel 3 and publish it as a local library, execute:
```

**Comment by schoeberl** [comment]
Thank you very much!

**Pull request merged**